### PR TITLE
Refactor solve form

### PR DESCRIPTION
**Pull request by thal0x**
Refactors form logic to remove a lot of left over tech debt from the previous non-unified API.

🎉  Also adds tests 🎉

### CODE REVIEW

File: src/components/SwapWidget/useSwapWidget.ts
Comment (bpiv400) [inline src/components/SwapWidget/useSwapWidget.ts:218]: Same comment I made for source assets. Also feels like this logic could be abstracted to a helper function since it's duplicated

File: src/components/TransactionDialog/TransactionDialogContent.tsx
Comment (bpiv400) [inline src/components/TransactionDialog/TransactionDialogContent.tsx:65]: Where are you getting pretty name from? LMK if this is coming from somewhere outside of the API

File: src/solve/execute-route.ts
Comment (bpiv400) [inline src/solve/execute-route.ts:79]: move to a helper function and move the hard code values to constants that can be discovered more easily and changed in one place

File: src/solve/execute-route.ts
Comment (bpiv400) [inline src/solve/execute-route.ts:130]: Use the gas calculation helper function here

File: src/solve/execute-route.ts
Comment (bpiv400) [inline src/solve/execute-route.ts:26]: This is a pretty massive function. It might be worth using a couple of helpers throughout to shorten it, so it's easier to understand what it's doing (and to de-dupe logic where possible)

File: src/solve/execute-route.ts
Comment (bpiv400) [inline src/solve/execute-route.ts:60]: good helper function opportunity

File: src/solve/execute-route.ts
Comment (bpiv400) [inline src/solve/execute-route.ts:102]: Also good helper opportunity

File: src/solve/execute-route.ts
Comment (bpiv400) [inline src/solve/execute-route.ts:231]: Move this number to a well named constant var

File: src/solve/execute-route.ts
Comment (bpiv400) [inline src/solve/execute-route.ts:177]: Why do these have custom logic? Do we need to highlight this to users in our docs?

File: src/solve/utils.ts
Comment (bpiv400) [inline src/solve/utils.ts:36]: im sorry for causing this

**Review by bpiv400 — CHANGES_REQUESTED** [review]

File: src/components/SwapWidget/useSwapWidget.ts
Comment (thal0x) [reply src/components/SwapWidget/useSwapWidget.ts:189]: This is already handled elsewhere

Comment (thal0x) [reply src/components/SwapWidget/useSwapWidget.ts:189]: https://github.com/skip-mev/ibc-dot-fun/blob/staging/src/context/assets.tsx#L47

File: src/components/TransactionDialog/TransactionDialogContent.tsx
Comment (zrbecker) [reply src/components/TransactionDialog/TransactionDialogContent.tsx:65]: Looks like it comes from chain-registry, our chains endpoint has a chain-name but it's not capitalized.

I assume he is using it this way because the chain registry one is easy to access.

**Comment by zrbecker** [comment]
Added a few comments, agree with Barry's comments.

**Review by zrbecker — CHANGES_REQUESTED** [review]
Let me know if you need another review on this before merge.

**Pull request merged**
